File: PreferredStock.py
from Stock import Stock

class PreferredStock(Stock):
    def __init__(self, symbol, lastDividend, fixedDividend, parValue):
        Stock.__init__(self, symbol, lastDividend, parValue)
        self.parValue = parValue
        
    def getDividendYield(self, price):
        return (self.fixedDividend * self.parValue) / self.price
        

File: C++/PreferredStock.cpp
/*
 * PreferredStock.cpp
 *
 *  Created on: 28/02/2016
 *      Author: oscar
 */

#include "PreferredStock.h"

PreferredStock::PreferredStock(Symbol symbol, unsigned lastDividend, unsigned parValue, unsigned fixedDividend) :
        Stock(symbol, lastDividend, parValue), m_fixedDividend(fixedDividend)
{
}

PreferredStock::~PreferredStock()
{
}

double
PreferredStock::getDividend() const
{
    return m_fixedDividend * m_parValue;
}
